### PR TITLE
[Feature request] Display version of guac-visualizer and backend GUAC

### LINKED ISSUE #62: [Feature request] Display version of guac-visualizer and backend GUAC
It can be unclear what version of the guac-visualizer and what version of GUAC I'm connected to. I don't know if I'm running against the latest released version, a dev version, etc.

### PR DESCRIPTION
**Pull request by nitram509**
resolves #62 

Changes:
* display guacgql server version in the footer, as well as visualizer version in the header.
* simplify logging code for backend calls
* send version information to guacgql backend

This is, how it looks like (look in the header, and footer)
Screenshot 2024-10-25 at 11 09 24

Hint:
The visualizer's version number is pulled from the package.json, which means when a release is made, this must be changed as well. In the Github Action, this is already implemented, see release.yaml the step named `Update version string to match git tag`

### CODE REVIEW

File: package.json
Comment (funnelfiasco) [inline package.json:6]: Out of curiosity, why the change from `main` to `0.0.0-development`? The version insertion in the release flow handles `main`. Is this a style preference (which is totally fine) or is there a somewhere else that needs a more version-looking format?

**Review by funnelfiasco — APPROVED** [review]
Looks good to me. The only thing I noticed was that when running with a version of GUAC that doesn't have a version endpoint (so anything prior to 0.10.2), the footer includes the HTML of the GraphQL playground. It's not worth blocking this for, since I doubt it will come up very often and it doesn't get in the way of using the visualizer.

**Review by lumjjb — APPROVED** [review]

File: package.json
Comment (nitram509) [reply package.json:6]: Indeed, rather a style preference, and three 'light' arguments:
* `0.0.0-development` is easier to (full-text-)search in the codebase
* `main` assumes higher knowledge that a specific git branch is meant. Also, it could be that a version was build from another branch, which then would be misleading.
* the proposed string is hopefully in a more version-ish format so more people realize it as such.

**Comment by nitram509** [comment]
@lumjjb may I ask to also set the `hacktoberfest-accepted` label, please?

File: package.json
Comment (funnelfiasco) [reply package.json:6]: Works for me!

**Comment by funnelfiasco** [comment]
@nitram509 the label is not necessary since this was merged. From the Hacktoberfest maintainer instructions:

> Be ready to review pull/merge requests, accepting those that are valid by merging them, leaving an overall approving review, or by adding the “hacktoberfest-accepted” label.

My Hacktoberfest dashboard shows 4/4 completed even though none of them have the label.

**Pull request merged**
